beetsplug/fetchart.py

```diff
@@ -610,13 +571,23 @@ def art_for_album(self, album, paths, local_only=False):
                 if self.maxwidth:
                     url = ArtResizer.shared.proxy_url(self.maxwidth, url)
                 candidate = self._fetch_image(url)
-                check = self._is_valid_image_candidate(candidate)
-                if check:
-                    out = candidate
-                    self._log.debug('using remote image {}', out)
-                    break
-
-        if self.maxwidth and out and check == CANDIDATE_DOWNSCALE:
+                if not candidate:
+                    continue
+
+                # get_size returns None if no local imaging backend is available
+                size = ArtResizer.shared.get_size(candidate)
+                self._log.debug('image size: {}', size)
+                if size:
+                    valid = ArtResizer.shared.valid_size(size,
+                        self.enforce_ratio, self.minwidth)
+                    if not valid:
+                        continue
+
+                out = candidate
+                self._log.debug('using remote image {}', out)
+                break
```

Comment:
Yeah, I've reverted a chunk of that, I think I overestimated the changes I'd need when I first started. Hopefully it's a bit cleaner now...